brave/brave-core
# Use Solana instruction backend decoder

Resolves https://github.com/brave/brave-browser/issues/25681

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

Scenario 1: System Program
1. create and fund a Solana wallet
2. visit https://pwgoom.csb.app/
3. connect wallet
4. initiate a "Sign and Send Transaction"
5. click details on the confirm transaction panel
6. Transaction parameters should be displayed (addresses and amount)

https://user-images.githubusercontent.com/30185185/220450361-4d122c87-7320-4318-ae9d-7375b312a4bf.mov


Scenario 2: Token Program
1. create and fund a Solana wallet
2. visit https://pwgoom.csb.app/
3. connect wallet
4. initiate a "Sign and Send SPL Token Transaction"
5. click details on the confirm transaction panel
7. Transaction parameters should be displayed (addresses and amount)


https://user-images.githubusercontent.com/30185185/220450399-a216c89e-137d-419c-b6e8-67262b5b64c6.mov



Scenario 3: Non-Decodable Instructions
1. create and fund a Solana wallet
2. visit https://jup.ag/swap/
3. Connect wallet
4. initiate a swap between tokens
5. click details on the confirm transaction panel
8. Raw Transaction info should be displayed for non-decodable instructions (program id and data)


https://user-images.githubusercontent.com/30185185/220450414-e88706f7-8904-4a23-9845-96a7af479cde.mov
